src/Maths/math.vector.ts

```diff
@@ -2237,6 +2314,46 @@ export class Vector4 {
         return this;
     }
 
+    /**
+    * Moves a value current towards target.
+    *
+    * This is essentially the same as Vector4.Lerp but instead the function will ensure that the speed never exceeds maxDelta.
+    * Negative values of maxDelta pushes the value away from target.
+    * @param current current value
+    * @param target target value
+    * @param maxDelta max distance to move
+    * @returns resulting value
+    */
+    public static MoveTowards(current: Vector4, target: Vector4, maxDelta: number): Vector4 {
+        const result: Vector4 = new Vector4(0.0, 0.0, 0.0, 0.0);
+        Vector4.MoveTowardsToRef(current, target, maxDelta, result);
+        return result;
+    }
+
+    /**
+    * Moves a value current towards target.
+    *
+    * This is essentially the same as Vector4.Lerp but instead the function will ensure that the speed never exceeds maxDelta.
+    * Negative values of maxDelta pushes the value away from target.
+    * @param current current value
+    * @param target target value
+    * @param maxDelta max distance to move
+    * @param result vector to set resulting value
+    */
+    public static MoveTowardsToRef(current: Vector4, target: Vector4, maxDelta: number, result: Vector4): void {

```

Comment:
Well do we want that inside the framework? It seems really niche and Math is imported everywhere so I'm very conservative with its size

cc @sebavan  for a second thought

Reply:
I come across this function alot when porting Unity style game mechanics. Just thought it would be helper for general vector math mechanics.

Reply:
Maybe could be separate module like movement ?